subprojects/core/src/integTest/groovy/org/gradle/internal/operations/logging/LoggingBuildOperationProgressIntegTest.groovy

```diff
@@ -184,11 +190,18 @@ class LoggingBuildOperationProgressIntegTest extends AbstractIntegrationSpec {
 
         then:
         10.times {  projectCount ->
+            def allExecutionOp = operations.only("Execute doLast {} action for :project-${projectCount}:all")
+            def allExecutionOpTaskProgresses = allExecutionOp.progress
+
             10.times { taskCount ->
                 def taskExecutionOp = operations.only("Task :project-${projectCount}:myTask$taskCount")
                 def classesTaskProgresses = taskExecutionOp.progress
                 def threadedTaskLoggingProgress = classesTaskProgresses.find { it.detailsType == LogEvent && it.details.message == "from :project-${projectCount}:myTask$taskCount task external thread" }
                 assert threadedTaskLoggingProgress.details.logLevel == 'LIFECYCLE'
+
+                // logging done from task-a logger during task-b execution will result in logging linked to task-b
```

Comment:
Is this the case when task-a is still running at that point?

Reply:
no, the coverage covers task-a being finished when task-b is executing. You think we need coverage for both running in parallel and task b uses tasks a logger? I cannot think of a reason why that so would do that an never see such a scenario in gradle builds. dealing with parallel task execution in general is already covered